conans/test/integration/command/source_test.py

```diff
@@ -277,3 +277,15 @@ def source(self):
         client.run("source .")
         assert "conanfile.py: Calling source()" in client.out
         assert "CACHE:MYCACHE!!" in client.out
+
+
+def test_source_python_requires():
+    c = TestClient(default_server_user=True)
+    c.save({"conanfile.py": GenConanfile("pytool", "0.1")})
+    c.run("export . ")
+    c.run("upload * -r=default -c")
+    c.run("remove * -c")
+
+    c.save({"conanfile.py": GenConanfile().with_python_requires("pytool/0.1")}, clean_first=True)
+    c.run("source . ")
+    assert "pytool/0.1: Not found in local cache, looking in remotes" in c.out
```

Comment:
Depending on what this test is supposed to check, I think there is an assert missing. The "looking in remotes" message is always displayed, including in the pathological case fixed in this PR.

The issue #14848 shows this in the log, where the "looking in remotes" is displayed just before the failure.

Reply:
The test would fail if Conan returned a non 0 exit code so the test is covering the case, but you're right, it might be nice to assert more output :)